--- a/Makefile	2017-11-21 17:37:18.808741579 +0200
+++ b/Makefile	2017-11-21 17:49:42.315132637 +0200
@@ -706,8 +706,9 @@
 # besides the one in which crash occurred; or SIGTRAP kills my program when
 # I set a breakpoint"
 ifeq ($(BR2_TOOLCHAIN_HAS_THREADS),y)
-	find $(TARGET_DIR)/lib/ -type f -name 'libpthread*.so*' | \
-		xargs -r $(STRIPCMD) $(STRIP_STRIP_DEBUG)
+	find $(TARGET_DIR)/lib/ -type f -name 'libpthread*.so*' \
+		-exec $(BR2_EXTERNAL_WPE_RPI_PATH)/tools/can-strip '{}' \; \
+		-print0 | xargs -0r $(STRIPCMD) $(STRIP_STRIP_DEBUG)
 endif

 # Valgrind needs ld.so with enough information, so only strip
